### DISCUSSION TITLE
Non-floating tagging panel

### DISCUSSION THREAD

**Discussion by quickfold** [freeplane/freeplane#1996] [Ideas]
I would like a docked or built-in option for tags that just provides the function of tagging nodes by drag-and-drop. The current floating window blocks parts of the map and feels like a temporary dialogue.

Dimitry wrote "Because Tag Category Manager and Tag Editor have "OK" and "Cancel" buttons, is seems natural to have them in dialogs." I understand this point. It is fine to keep the Tag Category Manager and the Tag Editor as separate floating panels. The user can manage tags and categories and then close the dialog.

But I want a non-dialog option to display current tags while I work on the map, for the primary purpose of having tags easily available to tag nodes, and in a way that they don't block any part of the map. Dimitry is right that the Tag Editor should be used to edit tags and then closed. I want a Tag panel with a different purpose that is for tagging nodes as I work, and not for editing or managing tags.

**Comment by Illegit** [comment]
Hi quickfold, 
1) I do not understand well the benefit of having the TagEditor in the tool panel because currently, we may open many of them contemporarily, and it is not very useful if they all are in the tool panel.
2) In my case, I see the tags in the node/nodes, therefore I use 99% of times the TagCategoryEditor that is floating, and I prefer it in the tool panel. With that, I understand how and what tag I got, missed, and where. Today the TCE is working quite nicely but the only annoying point is moving it when I need that part of the screen, or looking for it when I use 3 monitors.
Probably you think of something different than the actual TE?

**Comment by quickfold** [comment]
Hi Illegit,

I want a docked tag panel that I can use to tag nodes. It doesn't matter to me whether that panel can also edit nodes. Dimitri notes that the Tag Editor has 'ok' and 'cancel' buttons, which are usually features of a temporary dialog. So it doesn't make sense to him to have the Tag Editor docked, and I'm ok with that. If you want a docked Tag Editor and not just a tag panel, then you may want to make a case to Dimitri that the Tag Editor shouldn't have 'ok' and 'cancel' buttons, but should be a different GUI altogether.

**Comment by euu2021** [comment]
It is an interesting use case for an utility panel (https://github.com/euu2021/Freeplane_UtilityPanels). It would be nice to have:

- a search bar at the top of the panel, to filter the tags. Also, highlighting nodes in the map that contain tags that match the tags search results
- option to sort alphabetically or by frequency
- hovering over a tag in the panel shows an inspector panel with a list of nodes that contain that tag
- option to pin tags

What else?

From now on, I will have much more free time, so I may try to implement it.

**Reply by euu2021** [reply]
Also:

- Left clicking on a tag, in the panel, activates a filter that shows only the nodes that have that tag (as described at https://github.com/freeplane/freeplane/discussions/1993)

**Reply by euu2021** [reply]
Utility Panels 1.17 introduces a Tags panel. It's still a very basic version. When the user clicks the tag, it'd added to the selected node:

![java_OtfB4mI9w4](https://github.com/user-attachments/assets/7c891006-b3d9-4429-9832-cc874b7aceec)

I created a thread in the UtilityPanels repository to discuss features for the tags panel: https://github.com/euu2021/Freeplane_UtilityPanels/discussions/20
